File: Problem Set 2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    float L = (count_letters(text) / (float) count_words(text)) * 100;
    float S = (count_sentences(text) / (float) count_words(text)) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Print the grade level
    if(index <1.0)
    {
        printf("Before Grade 1\n");
    }
    else if(index >=16.0)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }

}

    int count_letters(string text)
    {
        int count = 0;
        for (int i = 0; i < strlen(text); i++)
        {
            if (isalpha(text[i]))
            {
                count++;
            }
        }
        return count;
    }

int count_words(string text)
{
    int count = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: Problem Set 6/sentimental-readability/readability.py
from cs50 import get_string

def print_grade(grade_index):
    if grade_index < 1:
        print('Before Grade 1\n')
    elif grade_index >= 1 and grade_index <= 16:
        print(f'Grade {grade_index}\n')
    else:
        print('Grade 16+\n')


def readability(letters, words, sentences):
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    rounded_index = round(index)
    return rounded_index


text = get_string('Text: ')

letters = 0
words = 0
sentences = 0

for word in text:
    for character in word:
        if letters == 1:
            words += 1
        if character.isalpha():
            letters += 1
        elif character.isspace():
            words += 1
        elif character == '.' or character == '?' or character == '!':
            sentences += 1


grade_index = readability(letters, words, sentences)
print_grade(grade_index)
